etc/xcc-router/src/lib.rs

```diff
@@ -209,19 +172,14 @@ impl Router {
         env::promise_return(promise_id);
     }
 
-    #[private]
     pub fn send_refund(&self) -> Promise {
-        let parent = self
-            .parent
-            .get()
-            .unwrap_or_else(|| env::panic_str("ERR_CONTRACT_NOT_INITIALIZED"));
-
+        let parent = self.require_parent_caller();
         Promise::new(parent).transfer(REFUND_AMOUNT)
     }
 }
 
 impl Router {
-    fn require_parent_caller(&self) {
+    fn require_parent_caller(&self) -> AccountId {
```

Comment:
I guess it's out of the scope of this PR, but it looks like a bad design of the method. 1. The name doesn't correspond to what it really does. 2. And looks like more than one responsibility.

Reply:
Good point. I have refactored it in https://github.com/aurora-is-near/aurora-engine/pull/867/commits/c847479fff268be0c45d881ecd3c3ce993411d8a